### PR TITLE
[MRG+1] added a new method to add multiple parameters in url

### PR DESCRIPTION
**Pull request by nasirhjafri**
added a new method to add multiple parameters in url

The new function named `add_or_replace_parameters` takes a dictionary of key-value parameters and update that to given url.

```
    >>> import w3lib.url
    >>> w3lib.url.add_or_replace_parameter('http://www.example.com/index.php', {'arg': 'v'})
    'http://www.example.com/index.php?arg=v'
    >>> w3lib.url.add_or_replace_parameter('http://www.example.com/index.php?arg1=v1&arg2=v2&arg3=v3',
    {'arg4': 'v4', 'arg3': 'v3new'})
    'http://www.example.com/index.php?arg1=v1&arg2=v2&arg3=v3new&arg4=v4'
    >>>
```
connects #117

### CODE REVIEW

**Comment by nasirhjafri** [comment]
@lopuhin Can you please review?

**Comment by nasirhjafri** [comment]
@kmike Can you please review it. Thanks.

File: w3lib/url.py [hunk 1]
```diff
@@ -200,6 +200,18 @@ def url_query_cleaner(url, parameterlist=(), sep='&', kvsep='=', remove=False, u
     return url
 
 
+def _add_or_replace_parameters(url, params):
+    parsed = urlsplit(url)
+    args = parse_qsl(parsed.query, keep_blank_values=True)
+
+    new_args = OrderedDict(args)
+    new_args.update(params)
+    new_args = [(k, v) for k, v in new_args.items()]
```
Comment (lopuhin) [inline w3lib/url.py]: Can ``new_args.items()`` be passed to ``urlencode`` directly?

Comment (nasirhjafri) [reply w3lib/url.py]: I went through the `urlencode` code and it looks like it accepts a dict argument directly. I've removed the loop(dict to tuple).

cc: @lopuhin

Comment (lopuhin) [reply w3lib/url.py]: @nasirhjafri oh nice, thanks! I didn't realise that ``urlencode`` so picky about the first argument, thanks for resolving this nicely.

**Comment by nasirhjafri** [comment]
Thanks, @lopuhin for reviewing. :)

**Pull request merged**